ansible-ipi-install/roles/installer/tasks/15_disconnected_registry_create.yml

```diff
@@ -0,0 +1,255 @@
+      firewalld:
+        port: "{{ registry_port }}/tcp"
+        permanent: yes
+        state: enabled
+        zone: "{{ item }}"
+      become: yes
+      with_items:
+      - libvirt
+      - public
+      when: firewall != "iptables"
+  
+    - name: Open registry port for iptables
+      iptables:
+        chain: INPUT
+        protocol: tcp
+        destination_port: "{{ registry_port }}"
+        jump: ACCEPT
+      become: yes
+      when: firewall == "iptables"
+  
+    - name: Create directory to hold the registry files
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+        mode: u=rwX,g=rX,o=rX
+        recurse: yes
+      loop:
+        - "{{ registry_dir_auth }}"
+        - "{{ registry_dir_cert }}"
+        - "{{ registry_dir_data }}"
+      become: true
+  
+    - name: Generate htpasswd entry
+      command: htpasswd -bBn {{ disconnected_registry_user }} {{ disconnected_registry_password }}
+      register: htpass_entry
+  
+    - name: Write htpasswd file
+      copy:
+        content: '{{ htpass_entry.stdout }}'
+        dest: "{{ registry_dir_auth }}/htpasswd"
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+        backup: yes
+        force: yes
+  
+    - name: Set disconnected_auth
+      set_fact:
+        disconnected_registry_up: "{{ disconnected_registry_user }}:{{ disconnected_registry_password }}"
+  
+    - name: Create registry auth for pullsecret
+      set_fact:
+        disconnected_auth: '  {"{{ ansible_fqdn }}:{{ registry_port }}": {"auth": "{{ disconnected_registry_up | b64encode }}", "email": "{{ ansible_user }}@redhat.com" } }'
+  
+    - name: Write auth for disconnected to registry host
+      copy:
+        content: "{{ disconnected_auth }}"
+        dest: "{{ ansible_env.HOME }}/{{ registry_auth_file }}"
+        backup: yes
+        force: yes
+  
+    - name: Write auth for disconnected to localhost
+      copy:
+        content: "{{ disconnected_auth }}"
+        dest: "{{ lookup ('env', 'PWD') }}/{{ registry_auth_file }}"
+        backup: yes
+        force: yes
+      delegate_to: localhost
+  
+    - name: append auth to pullsecret
+      shell: |
+        echo '{{ pullsecret }}' | jq -c \
+          '.auths += {{ disconnected_auth }}'
+      register: new_pullsecret
+  
+    - name: set pullsecret with new auth
+      set_fact:
+        pullsecret: "  {{ new_pullsecret.stdout }}"
+  
+    - name: generate ssl certs
+      shell: |
+        openssl req \
+          -newkey rsa:4096 \
+            -nodes \
+            -sha256 \
+            -keyout {{ registry_dir_cert }}/domain.key \
+            -x509 -days 365 \
+            -out {{ registry_dir_cert }}/domain.crt \
+            -subj "/C={{ cert_country }}/ST={{ cert_state }}/L={{ cert_locality }}/O={{ cert_organization }}/OU={{ cert_organizational_unit }}/CN={{ ansible_fqdn }}"
```

Comment:
have not tested but I think this will do what you want here:

```
- name: Generate an OpenSSL private key
  openssl_privatekey:
    path: {{ registry_dir_cert }}/domain.key
    size: 4096

- name: Generate an OpenSSL CSR
  openssl_csr:
    path: "{{ registry_dir_cert }}/domain.csr"
    privatekey_path: "{{ registry_dir_cert }}/domain.key"
    common_name: "{{ ansible_fqdn }}"
    country_name: "{{ cert_country }}"
    state_or_province_name: "{{ cert_state }}"
    locality_name: "{{ cert_locality }}"
    organization_name: "{{ cert_organization }}"
    organizational_unit_name: "{{ cert_organizational_unit }}"
    
    

- name: Generate a self signed OpenSSL Certificate
  openssl_certificate:
    path: {{ registry_dir_cert }}/domain.crt
    privatekey_path: {{ registry_dir_cert }}/domain.key
    csr_path: {{ registry_dir_cert }}/domain.csr
    provider: selfsigned
```

Reply:
working. thanks. in next commit